File: hhspec/prototype/.qa/shoot.py
import asyncio
from playwright.async_api import async_playwright
import os

PAGES = [
    ("special-occasions", "http://localhost:5173/special-occasions"),
    ("special-occasions-guest", "http://localhost:5173/special-occasions?sub=guest"),
    ("search-lace", "http://localhost:5173/search?q=lace"),
    ("search-empty", "http://localhost:5173/search?q=xyznomatch"),
    ("search-noquery", "http://localhost:5173/search"),
    ("style-gallery", "http://localhost:5173/style-gallery"),
    ("notfound", "http://localhost:5173/a-nonexistent-url"),
]

OUT_DIR = "/Volumes/MAC/workspace/dreamy/hhspec/prototype/.qa"

async def main():
    os.makedirs(OUT_DIR, exist_ok=True)
    async with async_playwright() as pw:
        browser = await pw.chromium.launch()
        context = await browser.new_context(viewport={"width": 1440, "height": 900}, device_scale_factor=1)
        for name, url in PAGES:
            page = await context.new_page()
            try:
                await page.goto(url, wait_until="networkidle", timeout=30000)
            except Exception as e:
                print(f"{name}: nav warning {e}")
            await page.wait_for_timeout(1100)
            # dismiss any Vite error overlay (other agents may not have created their views yet)
            try:
                await page.evaluate(
                    """() => {
                        document.querySelectorAll('vite-error-overlay').forEach(el => el.remove());
                    }"""
                )
            except Exception:
                pass
            # close region modal if visible
            try:
                btn = page.locator('button:has-text("Skip for now")').first
                if await btn.is_visible(timeout=400):
                    await btn.click()
                    await page.wait_for_timeout(300)
            except Exception:
                pass
            # close cookie banner
            try:
                btn = page.locator('button:has-text("Accept")').first
                if await btn.is_visible(timeout=400):
                    await btn.click()
                    await page.wait_for_timeout(200)
            except Exception:
                pass
            # again kill overlay if it re-appeared
            try:
                await page.evaluate(
                    """() => {
                        document.querySelectorAll('vite-error-overlay').forEach(el => el.remove());
                    }"""
                )
            except Exception:
                pass
            await page.wait_for_timeout(500)
            out = f"{OUT_DIR}/shop-{name}.png"
            await page.screenshot(path=out, full_page=True)
            print(f"OK {name} -> {out}")
            await page.close()
        await browser.close()

asyncio.run(main())

File: hhspec/prototype/.qa/shoot.js
const { chromium } = require('playwright')

const pages = [
  ['home', '/'], ['wedding-dresses', '/wedding-dresses'], ['special-occasion', '/special-occasion'],
  ['accessories', '/accessories'], ['outdoor-weddings', '/outdoor-weddings'], ['pdp', '/product/aurelia-gown'],
  ['pdp-bridesmaid', '/product/seabreeze-bridesmaid'], ['pdp-veil', '/product/cathedral-veil'],
  ['search', '/search?q=sage'], ['cart', '/cart'], ['checkout', '/checkout'], ['order-success', '/order-success'],
  ['login', '/account/login'], ['account', '/account'], ['account-orders', '/account/orders'],
  ['account-order-detail', '/account/orders/1001'], ['account-addresses', '/account/addresses'],
  ['account-wishlist', '/account/wishlist'], ['account-settings', '/account/settings'],
  ['inspiration', '/inspiration'], ['real-weddings', '/real-weddings'],
  ['real-wedding-detail', '/real-weddings/coastal-emma-james'], ['blog', '/blog'],
  ['blog-post', '/blog/outdoor-wedding-guide'], ['wedding-guides', '/wedding-guides'],
  ['about', '/about'], ['contact', '/contact'], ['faq', '/faq'], ['notfound', '/nope-404']
]

;(async () => {
  const b = await chromium.launch()
  const realErrors = []
  const imageErrors = []
  for (const [vpName, w, h] of [['desktop', 1440, 900], ['mobile', 390, 844]]) {
    const ctx = await b.newContext({ viewport: { width: w, height: h } })
    for (const [name, path] of pages) {
      const pg = await ctx.newPage()
      const errs = []
      pg.on('console', (m) => { if (m.type() === 'error' && !m.text().includes('status of 404')) errs.push(m.text()) })
      pg.on('pageerror', (e) => errs.push(e.message))
      pg.on('response', (r) => { if (r.status() === 404 && /\.(jpg|png|webp)/.test(r.url())) imageErrors.push(`${name}: ${r.url().split('/').pop()}`) })
      await pg.goto('http://localhost:5175' + path, { waitUntil: 'networkidle', timeout: 20000 })
      await pg.waitForTimeout(500)
      await pg.screenshot({ path: `.qa/${vpName}-${name}.png`, fullPage: vpName === 'desktop' })
      if (errs.length) realErrors.push({ vp: vpName, name, errs })
      await pg.close()
    }
    await ctx.close()
  }
  await b.close()
  console.log('REAL JS ERRORS:', JSON.stringify(realErrors))
  console.log('IMAGE 404s:', JSON.stringify([...new Set(imageErrors)]))
})()
